### ISSUE TITLE
Incorrect handling of property overriding in subclasses.

### ISSUE THREAD

**Issue by dcbaker** [PyCQA/pylint#844]
pylint: 1.5.4
astroid 1.4.4
python: 2.7.11, 3.3.6, 3.5.1

If a class wants to override the getter or setter of a property in the parent class, then pylint incorrectly marks it as a "no-member" error, the code runs and works as expected.

``` python
class Parent(object):
    def __init__(self):
        self.__thing = 'foo'

    @property
    def thing(self):
        return self.__thing


class Child(Parent):
    @Parent.thing.getter
    def thing(self):
        return super(Child, self).thing + '!'


print(Child().thing)
```

I believe that the corollary to this case is that the following case is invalid, but marked as valid by pylint:

``` python
class Parent(object):
    def __init__(self):
        self.__thing = 'foo'

    @property
    def thing(self):
        return self.__thing


class Child(Parent):
    @thing.getter
    def thing(self):
        return super(Child, self).thing + '!'


print(Child().thing)
```

**Comment by PCManticore** [comment]
True, but this seems like an edge right now. We might improve it though with Pylint / astroid 2.0.

**Comment by leynos** [comment]
This seems to suddenly have started affecting my Tornado RequestHandler subclass since installing pylint-1.5.5.  I get a slew of no-member errors from field accesses to objects returned by self.current_user (which is a property).  Downgrading to pylint-1.5.4 and astroid-1.4.4 fixes this.

**Comment by PCManticore** [comment]
Can you show me an example of how the RequestHandler subclass looks with respect to the current_user property? I'm trying to see if it is the same issue as this one.

**Comment by leynos** [comment]
An example of this:

```python
from tornado.web import RequestHandler

class User(object):

    def __init__(self):
        self.foo = 'foo'
        self.bar = 'bar'
        self.qux = 'qux'

class MyRequestHandlerSubclass(RequestHandler):

    def get_current_user(self):
        return User()

class TestRequestHandler(MyRequestHandlerSubclass):

    def get(self):

        print(self.current_user.foo)
```
The method `get_current_user()` is called by the `current_user` property and cached.

The following is displayed by `pylint`:

```
E: 19,14: Method 'current_user' has no 'foo' member (no-member)
```

**Comment by sscherfke** [comment]
With Tornado 6, calling methods like `tornado.web.RequestHandler.get_argument('name')` leads to `E1120: No value for argument 'default' in method call (no-value-for-parameter)` errors b/c of their [overload decorators](https://github.com/tornadoweb/tornado/blob/master/tornado/web.py)

**Comment by backbord** [comment]
I think this related to https://github.com/PyCQA/pylint/issues/1623 and thusly to https://github.com/PyCQA/pylint/issues/1581

**Comment by sscherfke** [comment]
A workaround for Tornado 6 is the following astroid brain:
```python
# astroid/brain/brain_tornado.py
"""
Astroid hooks for tornado.

"""
import astroid


def requesthandler_funcs():
    return astroid.parse(
        """
        from tornado.web import RequestHandler

        class RequestHandlerMock(RequestHandler):
            def get_argument(name, default=None, strip=True):
                return object()
        """
    )


astroid.register_module_extender(astroid.MANAGER, 'tornado.web', requesthandler_funcs)
```
